library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -158,14 +191,15 @@ protected boolean isReady() {
     return true;
   }
 
-  private void invokeRenderer(String text) {
+  private void invokeRenderer(SpannableStringBuilder text) {
     if (Util.areEqual(lastRenderedCaption, text)) {
       // No change.
       return;
     }
-    this.lastRenderedCaption = text;
+    final SpannableStringBuilder textToRender = text == null ? null : new SpannableStringBuilder(text);
```

Comment:
I need the subclass of CharSequence (SpannableStringBuilder) to carry the formatting information as well, and I create a new instance before sending it to the renderer to avoid concurrent modifications from this thread as now we operate on references. Am I missing something?